Test/gsapTests/Predictors/EventDrivenPredictorTests.cpp

```diff
@@ -296,4 +305,102 @@ namespace EventDrivenPredictorTests {
         bus.waitAll();
         Assert::AreEqual(1, listener.getCount(), "Predictor didn't produce prediction");
     }
+
+    void savePts() {
+        static Log& log = Log::Instance();
+        log.SetVerbosity(LOG_TRACE);
+        std::string src = "3701";
+        auto timestamp = MessageClock::time_point(MessageClock::duration(1535391267115000));
+        MessageBus bus(std::launch::deferred);
+        MessageCounter listener(bus, src, MessageId::BatteryEod);
+        ConfigMap config = createConfig();
+
+        std::unique_ptr<PrognosticsModel> model;
+        std::unique_ptr<LoadEstimator> loadEstimator;
+        std::unique_ptr<Observer> observer;
+        std::unique_ptr<Predictor> predictor;
+        std::vector<std::unique_ptr<IMessageProcessor>> eventListeners;
+        EventDrivenTrajectoryService trajService(bus,
+                                                 std::unique_ptr<TrajectoryService>(
+                                                     new TrajectoryService()),
+                                                 src);
+
+        auto& mfactory = PrognosticsModelFactory::instance();
+        model = mfactory.Create("Battery", config);
+        eventListeners.push_back(
+            std::unique_ptr<IMessageProcessor>(new PowerConverter(bus, config, src)));
+
+        auto& lefactory = LoadEstimatorFactory::instance();
+        loadEstimator = lefactory.Create("Const", config);
+
+        auto& ofactory = ObserverFactory::instance();
+        observer = ofactory.Create("UKF", *model, config);
+
+        auto& pfactory = PredictorFactory::instance();
+        predictor = pfactory.Create("MC",
+                                    *model,
+                                    *loadEstimator,
+                                    trajService.getTrajectoryService(),
+                                    config);
+
+        auto edObs = observer ? std::unique_ptr<EventDrivenObserver>(
+                                    new EventDrivenObserver(bus, std::move(observer), src))
+                              : nullptr;
+
+        auto edPred = predictor ? std::unique_ptr<EventDrivenPredictor>(
+                                      new EventDrivenPredictor(bus, std::move(predictor), src))
+                                : nullptr;
+
+        bus.publish(
+            std::shared_ptr<Message>(new EmptyMessage(MessageId::RouteStart, src, timestamp)));
+
+        auto time = timestamp + MessageClock::duration(50000000);
```

Comment:
This should use a duration with a clear name, e.g. `timestamp + std::chrono::seconds(50)`